lms/djangoapps/courseware/access.py

```diff
@@ -136,8 +136,9 @@ def has_access(user, action, obj, course_key=None):
     if not user:
         user = AnonymousUser()
 
-    if in_preview_mode():
-        if not bool(has_staff_access_to_preview_mode(user=user, obj=obj, course_key=course_key)):
+    # Preview mode is only accessible by staff.
+    if in_preview_mode() and course_key:
```

Comment:
We want to check that if a course is being accessed and  `in_preview_mode()` is true, it should only be accessed by staff. Having a `course_key` means we are checking access for entry in course. And also following method calls  `administrative_accesses_to_course_for_user(user, course_key)` and `is_masquerading_as_student(user, course_key)` need `course_key` in order to return intended results.
`in_preview_mode()` may return True if `course_key` is `None` but here we only want to enter into line 141 if there is a `course_key` (depicting that course access is being checked, and we don't want to give access to course if user is non-staff)